src/deploy.sh

```diff
@@ -184,73 +203,64 @@ function deploy_parser_options()
     options_values['REBOOT']=1
   fi
 
-  IFS=' ' read -r -a options <<< "$raw_options"
-  for option in "${options[@]}"; do
-    if [[ "$option" =~ ^(--.*|-.*|test_mode) ]]; then
-      if [[ "$enable_collect_param" == 1 ]]; then
-        options_values['ERROR']='expected paramater'
-        return 22
-      fi
-
-      case "$option" in
-        --remote)
-          options_values['TARGET']="$REMOTE_TARGET"
-          continue
-          ;;
-        --local)
-          options_values['TARGET']="$LOCAL_TARGET"
-          continue
-          ;;
-        --vm)
-          options_values['TARGET']="$VM_TARGET"
-          continue
-          ;;
-        --reboot | -r)
-          options_values['REBOOT']=1
-          continue
-          ;;
-        --modules | -m)
-          options_values['MODULES']=1
-          continue
-          ;;
-        --list | -l)
-          options_values['LS']=1
-          continue
-          ;;
-        --ls-line | -s)
-          options_values['LS_LINE']=1
-          continue
-          ;;
-        --uninstall | -u)
-          enable_collect_param=1
-          uninstall=1
-          continue
-          ;;
-        test_mode)
-          options_values['TEST_MODE']='TEST_MODE'
-          ;;
-        *)
-          options_values['ERROR']="$option"
+  eval "set -- $options"
+
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      --remote)
+        options_values['TARGET']="$REMOTE_TARGET"
+        shift
+        ;;
+      --local)
+        options_values['TARGET']="$LOCAL_TARGET"
+        shift
+        ;;
+      --vm)
+        options_values['TARGET']="$VM_TARGET"
+        shift
+        ;;
+      --reboot | -r)
+        options_values['REBOOT']=1
+        shift
+        ;;
+      --modules | -m)
+        options_values['MODULES']=1
+        shift
+        ;;
+      --list | -l)
+        options_values['LS']=1
+        shift
+        ;;
+      --ls-line | -s)
+        options_values['LS_LINE']=1
+        shift
+        ;;
+      --uninstall | -u)
+        uninstall=1
+        options_values['UNINSTALL']+="$2"
+        shift 2
+        ;;
+      --) # End of options, beginning of arguments
+        shift
```

Comment:
This is part of of getopt! It will separate dashed options and not dashed options with two dashes. For example

`kw  xpto1 --option --option2=xpto2 xpto3` will be converted to

`kw --option --option2 'xpto2' -- xpto1 xpto3`

So the two dashes must be handled in the case statement

Reply:
btw, let's change the `--uninstall=KERNEL` synstax to `--uninstall "KERNEL"`. I think this makes more sense from the getopt perspective.

Reply:
getopt handles both as valid input and meaning the same thing!

Reply:
To make it clearer. This case doesn't refer to the double dash between the argument name and option. It refers to the separation of arguments with options (which getopt has recognized) and arguments without options. Not this one:


```
kw --arg "option" -- xpto
   ^
   |
```

But rather this one:

```
kw --arg "option" -- xpto
                   ^
                   |
```